Symfony/CS/Tests/Fixer/Contrib/OrderedUseFixerTest.php

```diff
@@ -638,4 +653,723 @@ public function provide70Cases()
+        {
+            $bar = $foo->toArray();
+            /** @var ArrayInterface $bar */
+
+            return function () use ($bar, $foo) {};
+        }
+    }
+}
+
+namespace BlaRoo {
+
+    use Zoo\Bar;
+    use SomeClass;
+    use Foo\Zar\Baz;
+    use Symfony\Annotation\Template, Symfony\Doctrine\Entities\Entity;
+
+    class AnnotatedClass
+    {
+        /**
+         * @Template(foobar=21)
+         * @param Entity $foo
+         */
+        public function doSomething($foo)
+        {
+            $bar = $foo->toArray();
+            /** @var ArrayInterface $bar */
+
+            return function () use ($bar, $foo) {};
+        }
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+<?php
+
+namespace FooRoo {
+
+    use Foo\Bar\FooBar as FooBaz;
+    use Zoo\Bar as ZooBar, Zoo\Tar1;
+    use Foo\Bar;
+    use Foo\Zar\Baz;
+    use Symfony\Annotation\Template;
+    use Foo\Bar\Foo as Fooo, Foo\Bir as FBB;
+    use SomeClass;
+
+    $a = new Bar();
+    $a = new FooBaz();
+    $a = new someclass();
+
+    use Zoo\Tar2;
+
+    class AnnotatedClass
+    {
+        /**
+         * @Template(foobar=21)
+         * @param Entity $foo
+         */
+        public function doSomething($foo)
+        {
+            $bar = $foo->toArray();
+            /** @var ArrayInterface $bar */
+
+            return function () use ($bar, $foo) {};
+        }
+    }
+}
+
+namespace BlaRoo {
+
+    use Foo\Zar\Baz;
+    use Zoo\Bar;
+    use SomeClass;
+    use Symfony\Annotation\Template, Symfony\Doctrine\Entities\Entity;
+
+    class AnnotatedClass
+    {
+        /**
+         * @Template(foobar=21)
+         * @param Entity $foo
+         */
+        public function doSomething($foo)
+        {
+            $bar = $foo->toArray();
+            /** @var ArrayInterface $bar */
+
+            return function () use ($bar, $foo) {};
+        }
+    }
+}
+EOF;
+
+        $this->makeTest($expected, $input);
+    }
+
+    public function testByLengthFixWithComment()
+    {
+        OrderedUseFixer::configure(array(OrderedUseFixer::SORT_LENGTH));
+
+        $expected = <<<'EOF'
+The normal
+use of this fixer
+should not change this sentence nor those statements below
+use Zoo\Bar;
+use Foo\Bar;
+use Foo\Zar\Baz;
+
+<?php
+
+use Foo\Bar;
+use Zoo\Tar, SomeClass;
+use Foo\Zar\Baz;
+use Foo\Bir as FBB;
+use Zoo\Bar as ZooBar;
+use Foo\Bar\Foo as Fooo, Foo\Bar\FooBar /* He there */ as FooBaz;
+use /* FIXME */Symfony\Annotation\Template;
+
+$a = new Bar();
+$a = new FooBaz();
+$a = new someclass();
+
+use Symfony\Doctrine\Entities\Entity;
+
+class AnnotatedClass
+{
+    /**
+     * @Template(foobar=21)
+     * @param Entity $foo
+     */
+    public function doSomething($foo)
+    {
+        $bar = $foo->toArray();
+        /** @var ArrayInterface $bar */
+
+        return function () use ($bar, $foo) {};
+    }
+}
+EOF;
+
+        $input = <<<'EOF'
+The normal
+use of this fixer
+should not change this sentence nor those statements below
```

Comment:
we should sort by length, and for those of the same length, sort them alphabetically